### PR TITLE
feat: codegen for multiple public input columns 

### PR DESCRIPTION
**Pull request by Maddiaa0**
Codegen multiple public input column checks in the verifier

### CODE REVIEW

File: bberg/src/circuit_builder.rs [hunk 1]
```diff
@@ -140,6 +140,14 @@ impl CircuitBuilder for BBFiles {
         let serial_relations = map_with_newline(relations, execute_serial_transformation);
         let serial_lookups = map_with_newline(permutations, execute_serial_transformation);
 
+        // With futures
```
Comment (jeanmon) [inline bberg/src/circuit_builder.rs]: Did you duplicate the whole block above by mistake?

File: bberg/src/verifier_builder.rs [hunk 2]
```diff
@@ -47,21 +47,29 @@ impl VerifierBuilder for BBFiles {
...
+            )
+        };
+
+        let (public_inputs_check, evaluate_public_inputs) = if has_public_input_columns {
+            let inputs_check = public_cols
+                .iter()
+                .enumerate()
+                .map(|(i, col_name)| public_inputs_column_transformation(&col_name, i))
+                .collect::<String>();
+
```
Comment (jeanmon) [inline bberg/src/verifier_builder.rs:72]: Does not the code below be modified to support multiple columns? (e.g., evaluate_public_input_column )

**Review by jeanmon — APPROVED** [review]
LGTM. Please have a look at the comments.

File: bberg/src/circuit_builder.rs (hunk 1 above)
Comment (Maddiaa0) [reply bberg/src/circuit_builder.rs]: i think i dodgy rebased whoopsie

File: bberg/src/verifier_builder.rs (hunk 2 above)
Comment (Maddiaa0) [reply bberg/src/verifier_builder.rs:72]: It is run per column, so its ok

**Pull request merged**